compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -6072,6 +6075,11 @@ private boolean isPresent(Node node) {
 
     private boolean checkIfAnonymous(Node node) {
         SyntaxKind parentKind = node.parent().kind();
+        if (node.kind() == SyntaxKind.OBJECT_TYPE_DESC && parentKind == SyntaxKind.DISTINCT_TYPE_DESC) {
+            if (node.parent().parent() != null) {
+                return node.parent().parent().kind() != SyntaxKind.TYPE_DEFINITION;
```

Comment:
Can add a comment mentioning that what is doing here?

Reply:
Updated. 488b1bae2d3986a38bb9066a9ee032b65e6f1b7d